converter/SEMSinput.py

```diff
@@ -156,46 +159,140 @@ def process_row(row):
     sql = "select district_id, label from districts"
     districts = [{"id": r[0], "name": r[1]} for r in c.execute(sql).fetchall()]
 
+    # look for either-or contests, which have the same label and description
+    sql = "select label, contest_text from contests where type = '1' group by label, contest_text having count(*) = 2"
+    either_neither_labels = [r[0] for r in c.execute(sql).fetchall()]
+    
     # contests
     sql = "select contest_id, contests.label, type, contests.district_id, num_vote_for, num_write_ins, contest_text, party_id, districts.label from contests, districts where contests.district_id = districts.district_id"
     contests = [{
         "id": r[0],
         "section": r[8],
         "districtId": r[3],
-        "type": "candidate" if r[2] == "0" else "yesno",
+        "type": "candidate",
         "partyId": None if r[7] == "0" else r[7],
-        "title": r[6].split("\\n")[1],
+        "title": cleanup_text(r[6]).split("\n")[1],
         "seats": int(r[4]),
         "allowWriteIns": int(r[5]) > 0
-    } for r in c.execute(sql).fetchall()]
+    } if r[2] == "0" else {
+        "id": r[0],
+        "section": r[8],
+        "districtId": r[3],
+        "type": "yesno",
+        "title": cleanup_text(r[6]).split("\n")[0] + "\n" + r[1],
+        "description": "\n".join(cleanup_text(r[6]).split("\n")[1:])
+    } for r in c.execute(sql).fetchall() if r[1] not in either_neither_labels]
+
+    # either-or-contests
+    either_neither_contest_ids = []
+    for label in either_neither_labels:
+        sql = "select contest_id, contests.district_id, contest_text, districts.label, contests.label from contests, districts where contests.district_id = districts.district_id and contests.label = ? order by contest_id"
+        subcontests = c.execute(sql, [label]).fetchall()
+        text = cleanup_text(subcontests[0][2]).split("\n")
+        either_neither_contest_ids.append(subcontests[0][0])
+        either_neither_contest_ids.append(subcontests[1][0])
+        new_contest = {
+            "id": subcontests[0][0] + "-either-neither",
```

Comment:
Should we include both ids? No strong preference here.

Reply:
probably a good safe thing to do!